src/tile_layer.js

```diff
@@ -79,7 +79,14 @@ var TileLayer = L.TileLayer.extend({
 
         var templated = L.Util.template(url, tilePoint);
         if (!templated || !this.options.format) {
-            return templated;
+            var tileURL;
+            if (L.Browser.retina && url.indexOf('/styles/v1') !== -1) {
+                tileURL = [templated.slice(0, templated.indexOf('?')), '@2x', templated.slice(templated.indexOf('?'))].join('');
```

Comment:
Nit: this could probably be written in a simpler way, maybe `templated.replace('?', '@2x?')`?

Reply:
Wow, yep - would make this much more readable. Switched to `.replace` in https://github.com/mapbox/mapbox.js/pull/1317/commits/bd3b7bc38427863e16463b8e1f8bde65144d0c90